### ISSUE TITLE
Only auto-create sections for home and first level

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#6171]
I'm in the process of a simplification of how we create the "page tree" (and especially the section tree) in preparation for #6041 -- but also needed for the things coming ahead.

This is mostly code simplification which also should (probably) make it a little faster. But there is one part of the "section logic" that I will remove. Both because it is complicated and hard to maintain, but also because I suspect many people don't like the current behaviour, and it's kind of impossible to opt-out off.

Currently, with this tree:

```bash
content
├── a
│   └── b
│       └── c
│           └── d
│               └── _index.md
└── b
``` 

* Hugo will create a section for `a/b/c/d` because it has a "_index.*" content file.
* Hugo will also create sections for `a`, `a/b`, `a/b/c` and `b`.
* Hugo will also create a home page (aka the root section).

With the new simplified behaviour, Hugo will only "auto create" sections for the home page and the top level folders.

So in the example above, home page and `a` and `b` will be "auto created",  `a/b/c/d` will be a section because of its `_index.md` file.

Also related:

#6153
#6154

**Comment by kaushalmodi** [comment]
> So in the example above, home page and a and b will be "auto created", a/b/c/d will be a section because of its _index.md file.

Sounds fair.. so if one needs to retain the auto-creation of `a/b` and `a/b/c`, they simply need to create those two `_index.md` that I show below with the `†` sign, right?

```
content
├── a
│   └── b
│       ├── _index.md†
│       └── c
│           ├── _index.md†
│           └── d
│               └── _index.md
└── b
```

**Comment by kaushalmodi** [comment]
@bep So from your content tree structure:

```
content
├── a
│   └── b
│       └── c
│           └── d
│               └── _index.md
└── b
```

below is a good summary?

|                                                                                 | Created right now? | Will be created after this change?             |
|---------------------------------------------------------------------------------|--------------------|------------------------------------------------|
| Hugo will create a section for a/b/c/d, because it has a content file.          | Yes                | Yes                                            |
| Hugo will also create sections for a and b, because they are top level folders. | Yes                | Yes                                            |
| Hugo will also create a home page (aka the root section).                       | Yes                | Yes                                            |
| Hugo will also create sections for a/b and a/b/c.                               | Yes                | No, because they do not have any content file. |

**Comment by bep** [comment]
>Sounds fair.. so if one needs to retain the auto-creation of a/b and a/b/c, they simply need to create those two _index.md that I show below with the † sign, right?

Yes. Which is kind of "manual" and not automatic... But I think less magical, and it certainly removes a fair amount of hard-to-understand source code.

Your table is spot on.

**Comment by bep** [comment]
Ref. the discussion above, and esp. @kaushalmodi 's table: Does this change make sense? If you see some unforeseen consequences, please explain below. 

[![](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01DHM56YTB4ASY14Y1G1CQ16RN/Yes%2C%20that%20sounds%20good.)](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01DHM56YTB4ASY14Y1G1CQ16RN/Yes%2C%20that%20sounds%20good./vote)
[![](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01DHM56YTB4ASY14Y1G1CQ16RN/No%2C%20don%27t%20do%20that.)](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01DHM56YTB4ASY14Y1G1CQ16RN/No%2C%20don%27t%20do%20that./vote)

**Comment by bep** [comment]
This was fixed in 7ff0a8ee9fe8d710d407e57faf1fda43bd635f28
